php/libraries/Utility.class.inc

```diff
@@ -1308,5 +1308,45 @@ class Utility
         $resolvedPath = implode('/', $resolvedPath);
         return $resolvedPath;
     }
+
+    /**
+     * Get maximum upload size. Will be the smaller value of
+     * upload_max_filesize and post_max_size
+     *
+     * @return string smaller value of upload_max_filesize or post_max_size
+     */
+    static function getMaxUploadSize()
+    {
+        $umf = ini_get('upload_max_filesize');
+        $pms = ini_get('post_max_size');
+        return self::returnBytes($umf) < self::returnBytes($pms) ? $umf : $pms;
+    }
+
+    /**
+     * Takes in a storage string (e.g., 64K, 1024M, 2G, etc.)
+     * and returns value in bytes
+     *
+     * @param string $val The formatted string
+     *
+     * @return int|string The formatted string represented as bytes
+     *
+     * @note taken from http://php.net/manual/en/function.ini-get.php
+     */
+    static function returnBytes($val)
+    {
+        $val  = trim($val);
+        $last = strtolower($val[strlen($val)-1]);
+
+        switch($last) {
+        case 'g':
+                $val *= 1024;
+        case 'm':
+                $val *= 1024;
+        case 'k':
+                $val *= 1024;
+        }
```

Comment:
There's no reason to try and be clever and rely on the case falling through without documenting why/how it works. It would be much easier to understand if you just did something like:

```
        case 'g':
                return $val * 1024 * 1024 * 1024;
        case 'm':
                return $val * 1024 * 1024;
        case 'k':
                 return $val * 1024;
          default:
                  return $val;
```